### PR TITLE
Add links to CONTRIBUTING.md for all data items that can be contributed

### LINKED ISSUE #1573: Backlinks to sources
Note: this has been selected as an Outreachy issue. To be assigned, you must have completed the instructions in https://github.com/ocaml/ocaml.org/issues/1553.

# Task

Add contribute links on all pages that display data from the `data/` folder.

The data from the `data/` folder gets transformed to OCaml modules via the command line tool in `tools/ood-gen`.

Some pages (e.g. Documentation, Jobs) include links to their source in order to invite contributions. However, this is not the case for all the pages.

How to solve this issue:

- Go through all different kinds of data in [the CONTRIBUTING.md instructions](https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md) to check where they are displayed, and add a link to the respective section in the contributing instructions at https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md.

Here is an example of how it was done for the Jobs page: https://github.com/ocaml/ocaml.org/pull/1568

### PR DESCRIPTION
**Pull request by mays4**
Resolves #1573

This PR is to resolve the issue in #1573 and add contributing links on all pages that display data from the data/ folder.

### CODE REVIEW

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 1]
```diff
@@ -115,6 +122,12 @@ Layout.render
           <a class="btn btn-lg btn-tertiary" href="<%s Url.industrial_users %>">Success Stories</a>
         </div>
       </div>
+ 
+
     </div>
+   
+ <div class="prose max-w-none text-white pl-6 pb-6">
+    If you want to contribute to a new success story, check out this <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#story" target="_blank" style="color: white !important; text-decoration: none;"><strong>contributing guide</strong></a> on GitHub.
+</div>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]: I would move this to the industrial users page, since this is where the actual success stories are listed.

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 2]
```diff
@@ -95,7 +97,12 @@ Layout.render
         </div>
         <div class="grid lg:grid-cols-3 gap-8 lg:gap-16"></div>
       </div>
+       <div class="prose max-w-none text-white ml-2 mt-10 ">
+        If you want to contribute to add a new academic user, check out this <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#academic-user" target="_blank">contributing guide</a> on GitHub.
+  </div>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]: Probably better positioned right below the academic institution logos.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -87,33 +87,33 @@ The success stories should be structured in the following way:
...
+### <a name="content-book"></a><a name="books"> Add a Book</a>
 
 > Contribute to the [OCaml Books](https://ocaml.org/books).
 
 You can add a new OCaml book by creating a new Markdown file in [data/books/](data/books/). For instance: [ocaml-from-the-very-beginning.md](data/industrial_users/en/ocaml-from-the-very-beginning.md).
 
-### <a name="content-event"></a> Add an Event
+### <a name="content-event"></a><a name="event"> Add an Event</a>
 
 > Contribute to the [Community Events](https://ocaml.org/community).
```
Comment (sabine) [inline CONTRIBUTING.md]: This is technically incorrectly named. Best to rename "event" here to "meetup" because that's all this is right now.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -72,7 +72,7 @@ Please make sure that the job involves mostly writing OCaml. Contributions to ad
 
 If you notice that a job opportunity is outdated (e.g., already fulfilled or not open anymore), PRs to remove it are welcome as well.
 
-### <a name="content-success-story"></a> Add a Success Story
+### <a name="content-success-story"></a><a name="story"> Add a Success Story</a>
```
Comment (sabine) [inline CONTRIBUTING.md]:
```suggestion
### <a name="content-success-story"></a><a name="success-story"> Add a Success Story</a>
```

File: src/ocamlorg_frontend/pages/blog.eml [hunk 5]
```diff
@@ -89,5 +90,8 @@ Layout.render
                 <div><a href="<%s Url.news %>" class="link-orange">See all</a></div>
             </div>
         </div>
+        <div class="prose mt-8">
+            <p style="display: inline;">If you want to contribute a blog post, check out this <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#blog" target="_blank">Contribute Guide</a> on GitHub.</p>
+        </div>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/blog.eml]:
```suggestion
        <div class="prose mt-8">
            <p style="display: inline;">If you want to contribute a blog post, or add your RSS feed, check out the <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#blog" target="_blank">Contributing Guide</a> on GitHub.</p>
        </div>
```

Comment (sabine) [reply src/ocamlorg_frontend/pages/blog.eml]: maybe try and be consistent "check out the Contributing Guide".

File: src/ocamlorg_frontend/pages/packages.eml [hunk 6]
```diff
@@ -178,10 +178,20 @@ Layout.render
                     degree of metadata quality. This makes it one of the most stable package repositories available
                     today.</div>
             </div>
+      
         </div>
+         <div class="prose ml-2 mt-20">
+    <p style="display: inline;white-space: nowrap">If you want to contribute to update the list of featured packages, check out this <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#package" target="_blank">Contribute Guide</a> on GitHub.</p>
+</div>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/packages.eml]: We should not invite people to update the featured packages.

**Review by sabine — COMMENTED** [review]
@mays4 looks to be on the right track! :+1: I left some feedback.

File: src/ocamlorg_frontend/pages/packages.eml (hunk 6 above)
Comment (mays4) [reply src/ocamlorg_frontend/pages/packages.eml]: @sabine Thank you for your feedback. I have made adjustments to the title is the way I did correct?

File: src/ocamlorg_frontend/pages/blog.eml (hunk 5 above)
Comment (mays4) [reply src/ocamlorg_frontend/pages/blog.eml]: @sabine  Thank you for the suggestion. I have updated the text to 'check out the Contributing Guide' for consistency, ensuring that this style is maintained across all links, I also changed in jobs page that was given as an example.

File: CONTRIBUTING.md (hunk 3 above)
Comment (mays4) [reply CONTRIBUTING.md]: @sabine  I moved the link from the event page to the community page as this is the location where I see the meetup is that correct?

**Comment by mays4** [comment]
@sabine  Thank you for your valuable feedback. I've made the necessary adjustments to the positions of the links, and I would appreciate your confirmation that they are now correctly placed. I've added comments above in response to your feedback to ensure that I have understood your suggestions correctly.

Additionally, could you please verify if all the required links have been added or if I might have missed any? Your support has been greatly appreciated, and I would be grateful for any further input you might have. Please let me know when I can move the pull request from the draft, and if you have any more feedback, please don't hesitate to share. Thank you again for your assistance.

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 7]
```diff
@@ -13,6 +13,7 @@ Layout.render
       </p>
     </div>
   </div>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]:
```suggestion
```

unnecessary whitespace change

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 8]
```diff
@@ -58,6 +59,7 @@ Layout.render
         <div class="font-bold mt-8"><%s item.name %></div>
       </a>
       <% ); %>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]:
```suggestion
```

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 9]
```diff
@@ -66,7 +68,12 @@ Layout.render
...
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]:
```suggestion
```

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 10]
```diff
@@ -66,7 +68,12 @@ Layout.render
         <div class="font-bold mt-8"><%s item.name %></div>
       </a>
       <% ); %>
+      
     </div>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]:
```suggestion
```

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 11]
```diff
@@ -66,7 +68,12 @@ Layout.render
         <div class="font-bold mt-8"><%s item.name %></div>
       </a>
       <% ); %>
+      
     </div>
+     
+  </div>
+  <div class="prose max-w-none  p-8">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]:
```suggestion
  <div class="prose max-w-none p-8">
```

File: src/ocamlorg_frontend/pages/academic_users.eml [hunk 12]
```diff
@@ -95,7 +102,10 @@ Layout.render
         </div>
         <div class="grid lg:grid-cols-3 gap-8 lg:gap-16"></div>
       </div>
+      
+  
     </div>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/academic_users.eml]:
```suggestion
    </div>
```

File: src/ocamlorg_frontend/pages/blog.eml [hunk 13]
```diff
@@ -7,6 +7,7 @@ Layout.render
 <div class="bg-default dark:bg-dark-default py-10">
     <div class="container-fluid wide">
         <h1 class="sr-only">OCaml Blog</h1>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/blog.eml]:
```suggestion
```

File: src/ocamlorg_frontend/pages/blog.eml [hunk 14]
```diff
@@ -89,5 +90,8 @@ Layout.render
                 <div><a href="<%s Url.news %>" class="link-orange">See all</a></div>
             </div>
         </div>
+        <div class="prose max-w-none  mt-8">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/blog.eml]:
```suggestion
        <div class="prose max-w-none mt-8">
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -87,33 +87,33 @@ The success stories should be structured in the following way:
...
 
-### <a name="content-package"></a> Add a Featured Package
+### <a name="content-package"></a> <a name="package">Add a Featured Package</a>
 
 > Contribute to the [Featured Packages](https://ocaml.org/packages).
 
 To update the list of Featured Packages on the Packages page, you can update [data/packages.yml](data/packages.yml)
 
-### <a name="ocaml-changelog"></a> OCaml Changelog
+### <a name="ocaml-changelog"></a><a name="changleog"> OCaml Changelog</a>
```
Comment (sabine) [inline CONTRIBUTING.md]: Notice how all of these headings already have an anchor that can be used. Seems like it is not necessary to add a new one.

File: src/ocamlorg_frontend/pages/books.eml [hunk 16]
```diff
@@ -73,3 +73,8 @@ Learn_layout.single_column_layout
         </div>
     </div>
 </div>
+<div class="prose max-w-none ml-12">
+            If you want to add a new book, check out the <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#books" target="_blank">Contributing Guide</a> on GitHub.
+        
+</div>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/books.eml]:
```suggestion
</div>
```

File: src/ocamlorg_frontend/pages/changelog.eml [hunk 17]
```diff
@@ -71,6 +71,10 @@ Layout.render
         </article>
         <% ); %>
       </div>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/changelog.eml]:
```suggestion
```

File: src/ocamlorg_frontend/pages/community.eml [hunk 18]
```diff
@@ -160,7 +160,11 @@ Layout.render
               </div>
           </a>
           <% ); %>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/community.eml]:
```suggestion
```

File: src/ocamlorg_frontend/pages/community.eml [hunk 19]
```diff
@@ -160,7 +160,11 @@ Layout.render
               </div>
           </a>
           <% ); %>
+          
       </div>
+       <div class="prose max-w-none  mt-8">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/community.eml]:
```suggestion
       <div class="prose max-w-none mt-8">
```

File: src/ocamlorg_frontend/pages/industrial_users.eml [hunk 20]
```diff
@@ -186,7 +189,11 @@ the community and learn more about how they use OCaml."
       </div>
     </div>
   </div>
+  <div class="prose max-w-none  pl-12">
+    If you want to contribute to a new success story, check out the <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#story" target="_blank">Contributing Guide</a> on GitHub.
 </div>
+</div>
+
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/industrial_users.eml]: move one section higher up, below the Case Studies (these are the success stories).

File: src/ocamlorg_frontend/pages/community.eml [hunk 21]
```diff
@@ -160,7 +160,11 @@ Layout.render
...
+            If you want to contribute to add a new community event, check out the <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#meetup" target="_blank">Contributing Guide</a> on GitHub.
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/community.eml]:
```suggestion
            If you want to contribute to add a new community event, check out the <a href="https://github.com/ocaml/ocaml.org/blob/main/CONTRIBUTING.md#content-meetup" target="_blank">Contributing Guide</a> on GitHub.
```

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -87,33 +87,33 @@ The success stories should be structured in the following way:
...
 
-### <a name="content-book"></a> Add a Book
+### <a name="content-book"></a><a name="books"> Add a Book</a>
 
 > Contribute to the [OCaml Books](https://ocaml.org/books).
 
 You can add a new OCaml book by creating a new Markdown file in [data/books/](data/books/). For instance: [ocaml-from-the-very-beginning.md](data/industrial_users/en/ocaml-from-the-very-beginning.md).
 
-### <a name="content-event"></a> Add an Event
+### <a name="content-event"></a><a name="meetup"> Add an Event</a>
```
Comment (sabine) [inline CONTRIBUTING.md]:
```suggestion
### <a name="content-meetup"></a>Add a Meetup
```

File: src/ocamlorg_frontend/pages/industrial_users.eml [hunk 23]
```diff
@@ -186,7 +189,11 @@ the community and learn more about how they use OCaml."
       </div>
     </div>
   </div>
+  <div class="prose max-w-none  pl-12">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/industrial_users.eml]:
```suggestion
  <div class="prose max-w-none pl-12">
```

File: src/ocamlorg_frontend/pages/industrial_users.eml [hunk 24]
```diff
@@ -200,9 +207,14 @@ the community and learn more about how they use OCaml."
         <div class="font-bold mt-8"><%s item.name %></div>
       </a>
       <% ); %>
+     
     </div>
   </div>
+   
 </div>
+<div class="bg-default prose max-w-none  p-8">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/industrial_users.eml]:
```suggestion
<div class="bg-default prose max-w-none p-8">
```

**Review by sabine — COMMENTED** [review]
Hey @mays4, thanks that's very useful, almost ready to merge! :+1: 

There's just a few things to clean up:
1. Try to avoid introducing unnecessary whitespace because this makes it more likely for the patch to conflict with other ongoing work and harder to see what is exactly being changed.
2. remove the `_esy` folder.

I left some specific comments on the code, but at some point I stopped making change suggestions on whitespace.

**Comment by mays4** [comment]
Hi @sabine,

Thank you for the feedback. I've made the following modifications based on your suggestions:

1-Addressed the issue of whitespace by removing all the unnecessary spaces. The adjustments should now ensure a cleaner presentation.
2-Deleted the __esy folder. I believe this should remove it, but please let me know if any further action is required to remove it from GitHub.
3-Removed all extra Anctors. Initially, I included them following the example provided on the jobs page, but I've now rectified this.
4-Adjusted the description from "add a new community event " to "add a new community meetup event " to better align with the terminology used.
5-Rectified the reference to the OCaml Changelog, which was previously linked to the [Featured Packages]. It should now reflect the appropriate content accurately.
6-Relocated the Success Stories section and made other modifications to various links.
Please feel free to review the changes, and if there are any other adjustments you'd like me to make or if you need any further clarification, please don't hesitate to let me know.

**Review by sabine — APPROVED** [review]
LGTM

Thanks for the contribution @mays4! :+1:
